### PR TITLE
[MIG] purchase_supplierinfo_discount: migrate to version 9.0

### PR DESCRIPTION
**Pull request by jweste**
I didn't touch the "test" folder as I'm not sure at all how to deal with it.

### CODE REVIEW

File: product_supplierinfo_discount/models/purchase_order.py [hunk 1]
```diff
@@ -19,49 +19,30 @@
...
-            if pl_pinfo.min_quantity <= qty_in_product_uom:
-                res['value']['discount'] = pl_pinfo.discount
-            else:
-                break
-        return res
+    @api.onchange(
+        'product_id', 'product_qty', 'product_uom', 'partner_id',
+        'date_order', 'date_planned', 'name', 'price_unit', 'state')
+    def _compute_discount(self):
+        discount = 0
```
Comment (pedrobaeza) [inline product_supplierinfo_discount/models/purchase_order.py]: Don't change discount if not product_id.

File: product_supplierinfo_discount/models/purchase_order.py [hunk 2]
```diff
@@ -19,49 +19,30 @@
...
-            [('suppinfo_id', 'in', sinfos.ids)], order="min_quantity")
-        if 'value' not in res:
-            res['value'] = {}
-        for pl_pinfo in pl_pinfos:
-            if pl_pinfo.min_quantity <= qty_in_product_uom:
-                res['value']['discount'] = pl_pinfo.discount
-            else:
-                break
-        return res
+    @api.onchange(
```
Comment (pedrobaeza) [inline product_supplierinfo_discount/models/purchase_order.py:28]: Why do you need all these onchange fields?

File: product_supplierinfo_discount/models/purchase_order.py [hunk 3]
```diff
@@ -19,49 +19,30 @@
...
+        'product_id', 'product_qty', 'product_uom', 'partner_id',
+        'date_order', 'date_planned', 'name', 'price_unit', 'state')
+    def _compute_discount(self):
```
Comment (pedrobaeza) [inline product_supplierinfo_discount/models/purchase_order.py]: This is not a compute field, so don't use it this nomenclature. Start with `onchange_`.

File: product_supplierinfo_discount/models/purchase_order.py [hunk 4]
```diff
@@ -19,49 +19,30 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
-from openerp import models, api
+from openerp import models, api, fields
```
Comment (pedrobaeza) [inline product_supplierinfo_discount/models/purchase_order.py]: Sort this by alphabetic order

File: product_supplierinfo_discount/__openerp__.py [hunk 5]
```diff
@@ -22,9 +22,10 @@
 
 {
     "name": "Discounts in product supplier info",
-    "version": "8.0.1.0.0",
+    "version": "9.0.1.0.0",
     "author": "Serv. Tecnol. Avanzados - Pedro M. Baeza, "
               "Antiun Ingeniería S.L., "
+              "Julien Weste"
```
Comment (pedrobaeza) [inline product_supplierinfo_discount/__openerp__.py]: Comma at the end

File: product_supplierinfo_discount/__openerp__.py [hunk 6]
```diff
@@ -22,9 +22,10 @@
 
 {
     "name": "Discounts in product supplier info",
-    "version": "8.0.1.0.0",
+    "version": "9.0.1.0.0",
     "author": "Serv. Tecnol. Avanzados - Pedro M. Baeza, "
```
Comment (pedrobaeza) [inline product_supplierinfo_discount/__openerp__.py]: Please switch these two companies (Serv. Tecn... and Antiun) to the new one we have made: `Tecnativa`

File: product_supplierinfo_discount/models/purchase_order.py [hunk 7]
```diff
@@ -19,49 +19,30 @@
...
-        return res
+    @api.onchange(
+        'product_id', 'product_qty', 'product_uom', 'partner_id',
+        'date_order', 'date_planned', 'name', 'price_unit', 'state')
+    def _compute_discount(self):
+        discount = 0
+        if self.product_id:
+            # Look for a possible discount
+            qty_in_product_uom = self.product_qty
+            tday = fields.Date.today()
```
Comment (pedrobaeza) [inline product_supplierinfo_discount/models/purchase_order.py]: Call it `today`, as you only save one letter and it's less understandable

File: product_supplierinfo_discount/models/purchase_order.py [hunk 8]
```diff
@@ -19,49 +19,30 @@
...
+            sinfos = self.env['product.supplierinfo'].search(
+                [('product_tmpl_id', '=', self.product_id.product_tmpl_id.id),
```
Comment (pedrobaeza) [inline product_supplierinfo_discount/models/purchase_order.py]: Use `self.product_id.seller_ids.filtered()` instead, and you will propitiate a better inheritability (for example, there's other module that adds the supplierinfo at product level, other that use supplierinfo for customers). With this form, all of them will be compatible.

File: product_supplierinfo_discount/models/purchase_order.py [hunk 9]
```diff
@@ -19,49 +19,30 @@
...
+                 ('name', 'child_of', self.partner_id.
+                    commercial_partner_id.id),
+                 '|', ('date_start', '=', False), ('date_start', '<=', tday),
+                 '|', ('date_end', '=', False), ('date_end', '>=', tday),
+                 ], order="min_qty desc")
+            for sinfo in sinfos:
```
Comment (pedrobaeza) [inline product_supplierinfo_discount/models/purchase_order.py]: You can include this criteria in the filtered sentence, sort by min_qty, and select the last one (`[:-1]`).

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
Test folder is for testing that the module works correctly, so it's needed. If Travis is red, you will need to adapt it to the new data layout.

**Comment by jweste** [comment]
Thanks for the review! Here are some changes.
I'm finalizing the tests and will push them soon.

**Comment by jweste** [comment]
Here are the tests. They are running ok on my side.

**Comment by pedrobaeza** [comment]
I didn't see that there were a previous migration of the module (https://github.com/OCA/purchase-workflow/pull/272), and it's currently working, so I close this one. You should check next time first before making the effort.

**Pull request closed without merging**